ocrd/ocrd/workspace.py

```diff
@@ -248,8 +248,13 @@ def save_mets(self):
         log.info("Saving mets '%s'", self.mets_target)
         if self.automatic_backup:
             WorkspaceBackupManager(self).add()
+
+        mask = umask(0)
+        umask(mask)
+        mode = 0o666 & ~mask
```

Comment:
Shouldn't this use the permissions `self.mets_target` had _before_ (if it existed at all, otherwise `0o666`)?

Reply:
That would require additional code. In most (= all typical) cases the result would be the same. Therefore I don't think that is necessary.

Reply:
Sure, but then why do you care about the umask? You could just set it to 666 regardless.

There are use cases for this already BTW: when serving workspaces on the web (e.g. as part of a repository) the service daemon should usually not be allowed write access. Or on a cluster users might share their workspaces but disallow write access beyond the group.

Reply:
I use umask to get the default file mode for new files (typically `0622`). That's the same mode which we would have if we did not use `atomic_write`. And it fits the use cases which you describe. `0666` would allow write access for anyone which is not desired in most cases.

Reply:
> That's the same mode which we would have if we did not use `atomic_write`.

Not true. Python's `open` does not change the permissions if the file already existed, but your code (still) does.

> And it fits the use cases which you describe.

It happens to (if the umask is set that way). My point was that it's up to the _user_ to decide file permissions, and that should always be respected by programs like ours.

> `0666` would allow write access for anyone which is not desired in most cases.

Neither is reverting a file's permissions to the umask default.

Reply:
> Not true. Python's open does not change the permissions if the file already existed, but your code (still) does.

`os.open` does take umask into account: https://docs.python.org/3.5/library/os.html

> os.open(path, flags, mode=0o777, *, dir_fd=None)
Open the file path and set various flags according to flags and possibly its mode according to mode. When computing mode, the current umask value is first masked out.